src/__experimental__/components/date/date.spec.js

```diff
@@ -0,0 +1,333 @@
+import React from 'react';
+import moment from 'moment';
+import { mount } from 'enzyme';
+import Date from './date.component';
+import DatePicker from './date-picker.component';
+import Textbox from '../textbox';
+
+describe('Date', () => {
+  let wrapper;
+
+  describe('when the Component is rendered', () => {
+    it("then the Textbox component should be rendered as it's descendant", () => {
+      wrapper = render({});
+      expect(wrapper.find(Textbox).exists()).toBe(true);
+    });
+  });
+
+  describe('when the "value" prop is an empty string', () => {
+    const currentDate = getFormattedDate(moment());
+
+    it('then the input element value should be set to today', () => {
+      wrapper = render({ value: '' });
+      simulateBlurOnInput(wrapper);
+      expect(wrapper.find('input').prop('value')).toBe(currentDate);
+    });
+  });
+
+  describe('when "autoFocus" prop is defined', () => {
+    it("then component's input should be focused after render", () => {
+      wrapper = render({ autoFocus: true });
+      const input = wrapper.find('input');
+      const focusedElement = document.activeElement;
+      expect(input.instance()).toBe(focusedElement);
+    });
+  });
+
+  describe('when autoFocus prop is not defined', () => {
+    it('does not sets focus on the input', () => {
+      wrapper = render({});
+      const input = wrapper.find('input');
+      const focusedElement = document.activeElement;
+      expect(input.instance()).not.toBe(focusedElement);
+    });
+  });
+
+  describe('when the "focus" event is triggered on the input', () => {
+    let onFocusFn;
+
+    beforeEach(() => {
+      onFocusFn = jest.fn();
+    });
+
+    it('then onFocus prop should have been called', () => {
+      wrapper = render({ onFocus: onFocusFn });
+      simulateFocusOnInput(wrapper);
+      expect(onFocusFn).toHaveBeenCalled();
+    });
+
+    describe('with autoFocus property on the component', () => {
+      it('should not render a DatePicker component', () => {
+        wrapper = render({ onFocus: onFocusFn, autoFocus: true });
+        simulateFocusOnInput(wrapper);
+        expect(wrapper.find(DatePicker).exists()).toBe(false);
+      });
+    });
+
+    describe('without autoFocus property on the component', () => {
+      it('should render a DatePicker component', () => {
+        wrapper = render({ onFocus: onFocusFn });
+        simulateFocusOnInput(wrapper);
+        expect(wrapper.find(DatePicker).exists()).toBe(true);
+      });
+    });
+  });
+
+  describe('when the "blur" event is triggered on the input', () => {
+    let onBlurFn;
+
+    beforeEach(() => {
+      onBlurFn = jest.fn();
+      wrapper = render({ onBlur: onBlurFn });
```

Comment:
Define onBlurFn as a `const` with name `onBlur` for `render({ onBlur })`.

Reply:
a `const` will be shared between tests and that could lead to false positives in assertions.